server/src/main/java/org/opensearch/gateway/BaseGatewayShardAllocator.java

```diff
@@ -120,6 +141,12 @@ public abstract AllocateUnassignedDecision makeAllocationDecision(
         Logger logger
     );
 
+    public abstract ConcurrentMap<ShardRouting, AllocateUnassignedDecision> makeAllocationDecision(
+            Set<ShardRouting> shards,
+            RoutingAllocation allocation,
+            Logger logger
+    );
+
```

Comment:
Let's make this normal method, is there a need for existing PSA/RSA to implement this new method ? 

And add a default implementation like

```
HashMap<ShardRouting, AllocateUnassignedDecision> allocationDecisions = new HashMap<>();
        for (ShardRouting unassignedShard: shards){
            allocationDecisions.put(unassignedShard, makeAllocationDecision(unassignedShard, allocation, logger));
        }
        return allocationDecisions;
```